### PR TITLE
#164047022 user registration

### PR DESCRIPTION
**Pull request by mnswaleh**
## Description ##
Enables user to register to authors Haven. Creates a footer to be used in the application and updates, navbar, logo and site icon.

## Type of change ##
Please select the relevant option
- [ ] Bug fix(a non-breaking change which fixes an issue)
- [x] New feature(a non-breaking change which adds functionality)
- [ ] Breaking change(fix of feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested? ##
Please describe the tests that you ran to verify your changes
- [x] End to end test
- [x] Integration test
- [x] unit test

## Checklist: ##
- [x] My code follows the style guidelines of this project
- [x] I have linted my code prior to submission
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing tests pass locally with my changes

## Related Pivotal Tracker stories ##
[ #164047022](https://www.pivotaltracker.com/story/show/164047022)

## Reviewers ##
- @andela/olympian-devs

## Relevant screenshots: ##
Screenshot 2019-04-13 at 11 09 26

Screenshot 2019-04-14 at 19 16 17

### CODE REVIEW

**Review by Kibetchirchir — CHANGES_REQUESTED** [review]
Add .env.example for guidance on .env variables

**Review by Kibetchirchir — APPROVED** [review]

File: src/components/signup/register.js [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+
+  validateInput = (input) => {
+    const emailRegex = RegExp(/(^[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$)/);
+    const passwordRegex = RegExp(/^(?=.*[A-Za-z].*)(?=.*[0-9].*)[A-Za-z0-9]{8,}$/);
+    const usernameRegex = RegExp(/(^[a-zA-Z][A-Za-z0-9]{3,}$)/);
+    const formData = this.state.user;
+    const formErrors = this.state.formErrors;
+    switch(input){
+      case 'username':
+        formErrors.username = usernameRegex.test(formData.username) ? '' : 'username should be alphanumeric, start with letter and more than 3 characters';
```
Comment (Derrickkip) [inline src/components/signup/register.js]: You should use undefined instead of the empty string ''

File: src/components/signup/register.js [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+        break;
+      case 'email':
+        formErrors.email = emailRegex.test(formData.email) ? '' : 'invalid email address';
```
Comment (Derrickkip) [inline src/components/signup/register.js]: Same here

File: src/routers/Approuter.js [hunk 3]
```diff
@@ -1,14 +1,14 @@
-import React, { Component } from 'react';
+import React from 'react';
 import { BrowserRouter, Route, Switch } from 'react-router-dom';
-import Login from '../components/Login/Login';
```
Comment (winniekariuki) [inline src/routers/Approuter.js:3]: Kindly explain why you have removed the login route yet it was from the develop.

**Review by winniekariuki — COMMENTED** [review]
Nice work in ensuring that all checks are passing but kindly explain why the login route has been removed in your Approuter.js yet it was in the develop.

File: src/components/signup/register.js [hunk 4]
```diff
@@ -0,0 +1,267 @@
...
+          </Row>
+        </Container>
+        <div>
+          <br />
+          <br />
+          <br />
+          <br />
+          <br />
+          <br />
+          <br />
```
Comment (Derrickkip) [inline src/components/signup/register.js]: Why all these breaks? You can use CSS to achieve the desired style

File: src/routers/Approuter.js (hunk 3 above)
Comment (mnswaleh) [reply src/routers/Approuter.js:3]: I do not have a login file on my pr so I cannot import from that particular file @winniekariuki

**Review by JumaKahiga — APPROVED** [review]
It's working as expected. Good job.

**Pull request merged**